lib/rspec/page-regression/matcher.rb

```diff
@@ -9,10 +9,15 @@ module RSpec::PageRegression
 
       @responsive_filepaths = FilePaths.responsive_file_paths(RSpec.current_example, args)
 
-      opt = args.select { |k,_| RENDER_ARGS.include?(k) }
-
-      Renderer.render_responsive(page, @responsive_filepaths, opt)
-      @comparisons = @responsive_filepaths.map{ |filepaths| ImageComparison.new(filepaths) }
+      render_options = args.select { |k,_| RENDER_ARGS.include?(k) }
+      compare_options = {}
+      compare_options[:threshold] = args.fetch(:threshold, RSpec::PageRegression.threshold)
+      compare_options[:mode] = args.fetch(:mode, RSpec::PageRegression.mode)
+      compare_options[:include_rect] = args.fetch(:include_area, nil)
+      compare_options[:exclude_rect] = args.fetch(:exclude_rect, nil)
```

Comment:
@jekuno here's my reasoning for going with the way that I initially suggested:

- _Reasons for hardwired args for the common cases:_  

  * So that typical users of this gem don't also need to learn imatcher in order to use it
  * To  protect those users from possible future changes to imatcher args.

- _Reason to support_ `:imatcher` _hash arg:_ 
   *  To allow users who do want to take advantage of other features of imatcher to be able to do so.  
   * Regarding fear of imatcher args that are incompatible, I wouldn't worry about that:; any user sophisticated enough to use fancier imatcher features would be sophisticated enough.
   *  `:imatcher` would be in `ALLOWED_ARGS` so no further whitelisting would be needed.

- _Reasons **not** to hardcode all imatcher options_:  

     * It would require duplication of documentation (or send users to read imatcher doc)
     * It would be extra work to maintain, with specs needed for each arg.
     * It would add more work to update this gem whenever imatcher updates.
     * It introduces unneeded coupling between versions.  We actually *shouldn't* depend on `~> 0.1.9` we should depend on `~> 0.1`.  In that way that anyone using both imatcher and this gem would be able to get bug fixes and new features of imatcher without waiting for us to update this gem (note we have a bad track record of waiting a year to fix issues...!). 

       [@palkan I assume that you're not following the pointless (IMHO) semver rule that allows backwards-incompatible changes in minor releases if major release is 0.  If so, please update to 1.0!]

      * It leaves an opening for potential naming conflicts (in the future) between args native to this gem and imatcher arg.  E.g. supposing we decide to come up with an arg `:strict` that has some impact on how spec names are mapped to filenames, and later imatcher comes up with an arg `:strict` that affects how floating-point roundoff is dealt with -- then we'd need to hardcode the imatcher arg as `:imatcher_strict` or some other made-up name, adding more confusion for users and developers alike.


     * It introduces unneeded coupling between the gems.  I could easily imagine a future update to this gem that would support other image comparison libraries or let the user provide their own.  For that reason there should be a minimal set of hardcoded args likely common to all such libraries, and all the more esoteric args would go in a library-specific hash.

In summary, I still agree with my original proposal.   What do you think now? :)